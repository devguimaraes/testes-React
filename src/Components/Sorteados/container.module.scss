.container {
  position: relative;
  top: -55px;

  display: flex;
  flex-direction: column;

  align-items: center;
  justify-content: center;
  text-align: center;

  background-color: #fff9eb;

  padding-top: 40px;
  border-top: 2px solid #000000;
  border-radius: 64px 64px 0px 0px;

  @media (min-width: 1024px) {
    justify-content: center;
  }
}

.sessao {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.titulo {
  padding-bottom: 1rem;
  color: #4b69fd;
  font-family: 'Poppins';
  font-style: normal;
  font-weight: 600;
  font-size: 20px;
  line-height: 30px;
  text-align: center;
}

.input {
  width: 328px;
  height: 48px;

  background-color: #ffffff;
  border: 1px solid #000000;

  border-radius: 37.5px;
  filter: drop-shadow(2px 2px 0px #000000);

  text-align: center;
  font-size: 24px;
}

.paragrafo {
  font-family: 'Poppins';
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 27px;
  display: flex;
  align-items: center;
  text-align: center;

  color: #444444;

  margin: 32px 0px;
}

.btnSortear {
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 16px auto 46px auto;

  width: 155px;
  height: 59.53px;

  padding: 16px;
  gap: 10px;
  font-weight: 600;

  font-size: 16px;
  line-height: 24px;
  color: white;
  text-align: center;

  background: #fe652b;
  mix-blend-mode: normal;
  border: 2px solid #000000;
  box-shadow: 2px 2px 0px #000000;
  border-radius: 37.5px;
}

.amigoSecreto {
  font-weight: 400;
  font-size: 18px;
  line-height: 27px;

  color: rgba(254, 101, 43, 0.99);

  mix-blend-mode: normal;
}

.imgAviao {
  margin-top: 24px;
}
