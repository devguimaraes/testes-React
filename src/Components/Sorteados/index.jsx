import { useState } from 'react';
import { useListaDeParticipantes } from '../../State/Hooks/useListaDeParticipantes';
import { useResultadoSorteio } from '../../State/Hooks/useResultadoSorteio';
import styles from './container.module.scss';
import { BsFillDice5Fill } from 'react-icons/bs';

export const Sorteados = () => {
  const participantes = useListaDeParticipantes();

  const [participanteDaVez, setParticipanteDaVez] = useState('');
  const [amigoSecreto, setAmigoSecreto] = useState('');

  const resultado = useResultadoSorteio();

  const sortear = (evento) => {
    evento.preventDefault();

    if (resultado.has(participanteDaVez)) {
      setAmigoSecreto(resultado.get(participanteDaVez));
    }
  };

  return (
    <div className={styles.container}>
      <h2 className={styles.titulo}>Quem vai tirar o papelzinho?</h2>
      <section className={styles.sessao}>
        <form onSubmit={sortear}>
          <select
            className={styles.input}
            required
            name="participanteDaVez"
            id="participanteDaVez"
            placeholder="Selecione o seu nome"
            value={participanteDaVez}
            onChange={(evento) => setParticipanteDaVez(evento.target.value)}
          >
            {participantes.map((participante) => (
              <option key={participante}>{participante}</option>
            ))}
          </select>
          <p className={styles.paragrafo}>
            Clique em sortear para descobrir quem é seu amigo secreto!
          </p>
          <button className={styles.btnSortear}>
            <BsFillDice5Fill size={24} /> Sortear!
          </button>
        </form>
        {amigoSecreto && (
          <p className={styles.amigoSecreto} role="alert">
            {amigoSecreto}
          </p>
        )}
        <img
          src="/imagens/aviao.png"
          className={styles.imgAviao}
          alt="Avião de papel na cor verde"
        />
      </section>
    </div>
  );
};

export default Sorteados;
