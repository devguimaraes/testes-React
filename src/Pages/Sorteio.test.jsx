import { render } from '@testing-library/react';
import { RecoilRoot } from 'recoil';
import Sorteados from '../Components/Sorteados';

describe('Teste de snapshot tela Sorteio', () => {
  test('Deve ser renderizado corretamente', () => {
    const { container } = render(
      <RecoilRoot>
        <Sorteados />
      </RecoilRoot>
    );

    expect(container).toMatchSnapshot();
  });
});
