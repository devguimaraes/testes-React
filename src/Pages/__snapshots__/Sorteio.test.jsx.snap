// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de snapshot tela Sorteio Deve ser renderizado corretamente 1`] = `
<div>
  <div
    class="container"
  >
    <h2
      class="titulo"
    >
      Quem vai tirar o papelzinho?
    </h2>
    <section>
      <form>
        <select
          id="participanteDaVez"
          name="participanteDaVez"
          placeholder="Selecione o seu nome"
          required=""
        />
        <p>
          Clique em sortear para descobrir quem Ã© seu amigo secreto!
        </p>
        <button>
          Sortear
        </button>
      </form>
      
    </section>
  </div>
</div>
`;
